src/rebar_prv_dialyzer.erl

```diff
@@ -442,3 +450,30 @@ no_warnings() ->
 get_config(State, Key, Default) ->
     Config = rebar_state:get(State, dialyzer, []),
     proplists:get_value(Key, Config, Default).
+
+-spec collect_nested_dependent_apps([atom()]) -> [atom()].
+collect_nested_dependent_apps(RootApps) ->
+    Deps = lists:foldl(fun collect_nested_dependent_apps/2, sets:new(), RootApps),
+    sets:to_list(Deps).
+
+-spec collect_nested_dependent_apps(atom(), rebar_set()) -> rebar_set().
+collect_nested_dependent_apps(App, Seen) ->
+    case sets:is_element(App, Seen) of
+        true ->
+            Seen;
+        false ->
+            Seen1 = sets:add_element(App, Seen),
+            case code:lib_dir(App) of
+                {error, _} ->
+                    Seen1;
```

Comment:
We should error if an app doesn't exist. The application can't be started if a dependency does not exist.